### PR TITLE
Multile Doc fixes

### PR DESCRIPTION
**Pull request by ashcrow**
* Adds missing kubeauth api docs
* Adds ``commctl`` specific documentation

### CODE REVIEW

File: doc/commctl.rst [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+~~~~~~~~~~
+
+.. todo::
+
+    TODO
+
+Via RPM
+~~~~~~~
+
+If you want to roll your own RPM the spec file can be found in ``contrib/package/rpm/commissaire.spec``.
```
Comment (mbarnes) [inline doc/commctl.rst]: nit: Suggest a comma after RPM.

File: doc/commctl.rst [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+On RHEL/CentOS/Fedora based systems you will also need to make sure to have an rpm build environment set up.
```
Comment (mbarnes) [inline doc/commctl.rst]: nit: Suggest capitalizing "rpm", unless you meant `rpmbuild`-the-tool.

File: doc/commctl.rst [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+
+For further dependencies please see ``BuildRequires`` in the spec file.
+
+
+Configuration
+-------------
+``commctl`` requires a configuration file. The default path is
+``~/.commissaire.json`` though it can be changed with the ``--config``/``-c``
+option.
+
```
Comment (mbarnes) [inline doc/commctl.rst]: Maybe mention the configuration minimally requires one endpoint?

File: doc/commctl.rst [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+If you are using the :ref:`Kubernetes authentication plugin <kubeauth>` you can opt to reuse the credentials from your kubeconfig like so:
+
+.. note::
+
+    If you include username/password and kubeconfig items the username/password will be ignored in favor of the kubeconfig.
+
+.. include:: examples/commctl_config_with_kubeconfig.rst
+
+Multiple endpoints may be specified. If the first endpoint is unreachable
+the next endpoint in the list is used.
```
Comment (mbarnes) [inline doc/commctl.rst]: Possibly clearer (I'll let you decide):
Multiple fallback endpoints may be specified as a list.  The endpoints are tried in order until a successful connection is made.

File: doc/commctl.rst [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+
+
+Commands
+--------
+
+cluster
+~~~~~~~
+
+create
+``````
```
Comment (mbarnes) [inline doc/commctl.rst]: This whole section feels like it wants to be a man page.  Would it help to more formally show the command syntax so it's a bit less wordy?  e.g.
```
commctl cluster create [OPTIONS...] NAME

NAME : The name to give the cluster

OPTIONS
-t / --type : ...
-n / --network : ...
```
Also, we should enumerate the valid choices where applicable, like for `--type`.

File: doc/commctl.rst [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+
+.. include:: examples/commctl_list_clusters.rst
+
+To list all hosts:
+
+.. include:: examples/commctl_list_hosts.rst
+
+To list all hosts in a specific cluster:
+
+.. include:: examples/commctl_list_hosts_in_cluster.rst
```
Comment (mbarnes) [inline doc/commctl.rst]: The bits about listing hosts should be moved to the `host` / `list` section.

File: doc/commctl.rst [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+
+deploy start
+````````````
+
+``deploy start`` will create a new deployment on an Atomic host.
```
Comment (mbarnes) [inline doc/commctl.rst]: Should mention this operation happens asynchronously (hence `start`) and that `deploy status` can be used to monitor progress.  Similarly for the other `start` commands.

File: doc/commctl.rst (hunk 5 above)
Comment (mbarnes) [inline doc/commctl.rst]: Another thought: would it be useful to link to the REST endpoint that each command wraps?  The endpoint section should be where the nitty gritty behavior descriptions live (imo).

File: doc/operations.rst [hunk 8]
```diff
@@ -4,40 +4,8 @@ Operations
 Preface
 -------
 All operations via commissaire are done via REST. While any HTTP client can
-be used this document will show examples using *commctl* as well as *curl*.
+be used this many users will feel more comfortable using :ref:`commctl <commctl_preface>`.
```
Comment (mbarnes) [inline doc/operations.rst]: I think `be used this many users...` is supposed to be `be used, many users...`

File: doc/commctl.rst (hunk 5 above)
Comment (ashcrow) [inline doc/commctl.rst]: I agree and think we should autogenerate the man page and the content in these locations. For now I just wrote it out so we had something.

For the cross-referencing, I'll go ahead and do that now.

File: doc/commctl.rst (hunk 6 above)
Comment (ashcrow) [inline doc/commctl.rst]: Oops :smile:

File: doc/commctl.rst (hunk 7 above)
Comment (ashcrow) [inline doc/commctl.rst]: Good idea.

File: doc/commctl.rst (hunk 4 above)
Comment (ashcrow) [inline doc/commctl.rst]: :+1:

File: doc/commctl.rst (hunk 3 above)
Comment (ashcrow) [inline doc/commctl.rst]: Sure.

File: doc/commctl.rst (hunk 2 above)
Comment (ashcrow) [inline doc/commctl.rst]: :+1:

File: doc/commctl.rst (hunk 1 above)
Comment (ashcrow) [inline doc/commctl.rst]: :+

**Comment by ashcrow** [comment]
:arrow_up:

**Comment by mbarnes** [comment]
Looks good!  I don't suppose this will work, but...

@rh-atomic-bot r+

**Comment by mbarnes** [comment]
Didn't think so.  Merged manually.

**Pull request closed without merging**